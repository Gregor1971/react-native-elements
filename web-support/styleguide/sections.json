[
  {
    "name": "Buttons",
    "components": "../../src/buttons/[A-Z]*.js"
  },
  {
    "name": "Cards",
    "components": "../../src/+(card|pricing)/[A-Z]*.js"
  },
  {
    "name": "Forms",
    "components": "../../src/form/[A-Z]*.js"
  },
  {
    "name": "Icons",
    "components": "../../src/+(icons|social)/[A-Z]*.js"
  },
  {
    "name": "Inputs",
    "components": "../../src/+(checkbox|rating|slider)/[A-Z]*.js"
  },
  {
    "name": "Lists",
    "components": "../../src/list/[A-Z]*.js"
  },
  {
    "name": "Tiles",
    "components": "../../src/tile/[A-Z]*.js"
  },
  {
    "name": "misc",
    "components":
      "../../src/+(avatar|badge|divider|header|text)/+(badge|[A-Z])*.js"
  }
]
