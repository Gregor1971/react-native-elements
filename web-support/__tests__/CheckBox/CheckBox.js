import parseSamples from 'enzyme-styleguidist-sample-parser';
import sections, { options } from '../../samples';

parseSamples({ CheckBox: sections.CheckBox }, options);
