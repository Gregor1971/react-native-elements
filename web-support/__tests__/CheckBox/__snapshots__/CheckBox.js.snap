// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox: CheckBox: [ prop: center, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: component, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableWithoutFeedback
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title
    </TextElement>
  </View>
</TouchableWithoutFeedback>
`;

exports[`CheckBox: CheckBox: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      Object {
        "backgroundColor": "#071",
      },
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: iconRight, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title
    </TextElement>
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: onLongPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        Object {
          "justifyContent": "flex-end",
        },
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: right, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        Object {
          "justifyContent": "flex-end",
        },
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`CheckBox: CheckBox: [ prop: textStyle & title ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          Object {
            "backgroundColor": "#071",
          },
          undefined,
        ]
      }
    >
      title
    </TextElement>
  </View>
</TouchableOpacity>
`;
